AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/Federation/ClearFederationToIdentityPool.swift

```diff
@@ -18,8 +18,18 @@ struct ClearFederationToIdentityPool: Action {
 
         let event: StateMachineEvent
         do {
-            try await credentialStoreClient?.deleteData(type: .amplifyCredentials)
-            event = AuthenticationEvent.init(eventType: .clearedFederationToIdentityPool)
+            // Check if we have credentials are of type federation, otherwise throw an error
+            if case .amplifyCredentials(let credentials) = try await credentialStoreClient?.fetchData(
```

Comment:
This action should not be called if there is not federated credentials right?

Reply:
I think this is to clear an error state?

Reply:
Had a discussion offline. 

>  I can avoid using the action to determine clearFederation is needed or not, by utilizing the credentials associated with the session error... But there could be a situation, where we don't get a sessionError and the user won't be able to clear..